pkg/innerring/internal/blockchain/blockchain.go

```diff
@@ -0,0 +1,608 @@
+	switch {
+	case neoNodeConfigFilepath == "":
+		panic("missing Neo node config file")
+	case acc == nil:
+		panic("missing node account")
+	case acc != nil && !acc.CanSign():
+		panic("account can not sign anything")
+	case log == nil:
+		panic("missing logger")
+	case chErr == nil:
+		panic("missing external error channel")
+	}
+
+	// TODO: (almost) all code below is copy-pasted from neo-go server command, consider sharing
+
+	cfg, err := config.LoadFile(neoNodeConfigFilepath)
+	if err != nil {
+		return nil, fmt.Errorf("load neo-go node config from file: %w", err)
+	}
+
+	cfgServer, err := network.NewServerConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("compose neo-go server config from the base one: %w", err)
+	}
+
+	blockchainStorage, err := storage.NewStore(cfg.ApplicationConfiguration.DBConfiguration)
+	if err != nil {
+		return nil, fmt.Errorf("init blockchain storage: %w", err)
+	}
+
+	blockChain, err := core.NewBlockchain(blockchainStorage, cfg.Blockchain(), log)
+	if err != nil {
+		closeErr := blockchainStorage.Close()
+		if closeErr != nil {
+			return nil, fmt.Errorf("init core blockchain component: %w; failed to close blockchain storage: %v", err, closeErr)
+		}
+
+		return nil, fmt.Errorf("init core blockchain component: %w", err)
+	}
+
+	go blockChain.Run() // is it required to be insta called?
+
+	netServer, err := network.NewServer(cfgServer, blockChain, blockChain.GetStateSyncModule(), log)
+	if err != nil {
+		return nil, fmt.Errorf("init neo-go network server: %w", err)
+	}
+
+	rpcActor := newActor(blockChain, netServer, cfg.ApplicationConfiguration.RPC.MaxGasInvoke)
+
+	_actor, err := actor.NewSimple(rpcActor, acc)
+	if err != nil {
+		return nil, fmt.Errorf("init simple actor using node account: %w", err)
+	}
+
+	stateRootMode := blockChain.GetStateModule().(*corestateroot.Module)
+
+	stateRootService, err := stateroot.New(cfgServer.StateRootCfg, stateRootMode, log, blockChain, netServer.BroadcastExtensible)
+	if err != nil {
+		return nil, fmt.Errorf("init StateRoot service: %w", err)
+	}
+
+	netServer.AddExtensibleService(stateRootService, stateroot.Category, stateRootService.OnPayload)
+
+	// do we need oracle service?
+
+	// var oracleService *oracle.Oracle
+	//
+	// if cfg.ApplicationConfiguration.Oracle.Enabled {
+	// 	var c oracle.Config
+	// 	c.Log = log
+	// 	c.Network = cfg.ProtocolConfiguration.Magic
+	// 	c.MainCfg = cfg.ApplicationConfiguration.Oracle
+	// 	c.Chain = artifacts.blockchain
+	// 	c.OnTransaction = artifacts.netServer.RelayTxn
+	//
+	// 	oracleService, err = oracle.NewOracle(c)
+	// 	if err != nil {
+	// 		return fmt.Errorf("init Oracle module: %w", err)
+	// 	}
+	//
+	// 	artifacts.blockchain.SetOracle(oracleService)
+	// 	artifacts.netServer.AddService(oracleService)
+	// }
+
+	var consensusService consensus.Service
+
+	if cfg.ApplicationConfiguration.Consensus.Enabled {
```

Comment:
Not really needed (the flag and unlock logic), you know when your IR is an alphabet node and CN uses the same key.